stdlib/InteractiveUtils/test/runtests.jl

```diff
@@ -643,7 +646,7 @@ end
 # macro options should accept both literals and variables
 let
     opt = false
-    @test !(first(@code_typed optimize=opt sum(1:10)).inferred)
+    first(@code_typed optimize=opt sum(1:10))
```

Comment:
```suggestion
    @test length(first(@code_typed optimize=opt sum(1:10)).code) ==
          length((@code_lowered sum(1:10)).code)
```
maybe?

Reply:
I don't think that invariant is guaranteed to hold. Inference does some cleanup

Reply:
I think it is supposed to hold, since inference does not need to mutate the IR. I think we were moving slowly towards a vision where inference can be assumed to be non-mutating entirely (so copies are not necessary just to run inference), though I don't know if we are there now

Reply:
Alright, I'll add the assertion and if it starts failing, I'll assign @aviatesk to finish that work ;)